test/isochrone.cc

```diff
@@ -0,0 +1,114 @@
+      "hov": {"max_distance": 5000000.0,"max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50},
+      "isochrone": {"max_contours": 4,"max_distance": 25000.0,"max_locations": 1,"max_time": 120},
+      "max_avoid_locations": 50,"max_radius": 200,"max_reachability": 100,
+      "multimodal": {"max_distance": 500000.0,"max_locations": 50,"max_matrix_distance": 0.0,"max_matrix_locations": 0},
+      "pedestrian": {"max_distance": 250000.0,"max_locations": 50,"max_matrix_distance": 200000.0,"max_matrix_locations": 50,"max_transit_walking_distance": 10000,"min_transit_walking_distance": 1},
+      "skadi": {"max_shape": 750000,"min_resample": 10.0},
+      "trace": {"max_distance": 200000.0,"max_gps_accuracy": 100.0,"max_search_radius": 100,"max_shape": 16000,"max_best_paths":4,"max_best_paths_shape":100},
+      "transit": {"max_distance": 500000.0,"max_locations": 50,"max_matrix_distance": 200000.0,"max_matrix_locations": 50},
+      "truck": {"max_distance": 5000000.0,"max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50}
+    }
+  })");
+
+} // namespace
+
+void try_isochrone(GraphReader& reader,
+                   loki_worker_t& loki_worker,
+                   thor_worker_t& thor_worker,
+                   const char* test_request,
+                   const std::string& expected) {
+  valhalla::valhalla_request_t request;
+  request.parse(test_request, valhalla::odin::DirectionsOptions::isochrone);
+  loki_worker.isochrones(request);
+
+  // Process isochrone request
+  auto result = thor_worker.isochrones(request);
+
+  // Check if the result contains the expected string
+  if (result.find(expected) == std::string::npos) {
+    throw std::runtime_error("isochrones failed: expected " + expected);
+  }
+}
+
+void test_isochrones() {
+  // Test setup
+  loki_worker_t loki_worker(config);
+  thor_worker_t thor_worker(config);
+  GraphReader reader(config.get_child("mjolnir"));
+
+  // Test auto isochrone with one contour
+  std::string expected1 = "\"type\":\"LineString\"";
+  const auto test_request1 =
+      R"({"locations":[{"lat":52.078937,"lon":5.115321}],"costing":"auto","contours":[{"time":15}],"polygons":false,"denoise":0.2,"generalize":150})";
+  try_isochrone(reader, loki_worker, thor_worker, test_request1, expected1);
+
+  // Try pedestrian isochrone with one contour, polygon=true
+  std::string expected2 = "\"type\":\"Polygon\"";
+  const auto test_request2 =
+      R"({"locations":[{"lat":52.078937,"lon":5.115321}],"costing":"bicycle","contours":[{"time":15}],"polygons":true,"denoise":0.2})";
+  try_isochrone(reader, loki_worker, thor_worker, test_request2, expected2);
+}
+
+int main(int argc, char* argv[]) {
+  test::suite suite("trivial_paths");
+
+  // Silence logs (especially long request logging)
+  logging::Configure({{"type", ""}});
+
+  suite.test(TEST_CASE(test_isochrones));
+
+  return suite.tear_down();
+}
```

Comment:
Shouldn't we add a few date/time tests in here as well?

Reply:
I think we need to wait until we have a test data set for Utrecht with predicted speed data. Also, not sure how to really test the results yet. There were no isochrone unit tests so I added this as a start.